### DISCUSSION TITLE
CI/CD and various scripts in `.github/scripts`

### DISCUSSION THREAD

**Discussion by VeckoTheGecko** [Parcels-code/Parcels#1276] [Q&A] [answered]
I was looking through the CI/CD workflows for the project, and came across the calls to the installation and testing scripts in the [`.github/scripts` folder](https://github.com/VeckoTheGecko/parcels/tree/master/.github/scripts) for the various platforms.

It feels a bit strange having many scripts for this setup, rather than just invoking the [Setup Miniconda](https://github.com/marketplace/actions/setup-miniconda) workflow with the appropriate environment file. I think using this workflow would greatly simplify the CI/CD eliminating the use for the custom scripts (and would also make for a good example for fellow researchers to copy to get started with unit testing and CI/CD in their parcels projects).

My questions are:
- Is the use of these scripts just for the running of CI/CD? (i.e. don't play a role in building the package, or aren't called elsewhere in the codebase?)
- I'm unfamiliar with some of the stuff happening in some of the scripts ([allowing for a plotting display??](https://github.com/VeckoTheGecko/parcels/blob/56e988ae0f627ec8994c656f5f4bab71154a2e47/.github/scripts/linux_integrationtest.sh#L5-L8), [custom variables being set](https://github.com/VeckoTheGecko/parcels/blob/56e988ae0f627ec8994c656f5f4bab71154a2e47/.github/scripts/osx_unittest.sh#L4-L5)). Could you shed light on these items?

**Comment by erikvansebille** [comment, marked as answer]
Thanks for the question, @VeckoTheGecko. The CI was set up for us by IT support from Utrecht University (specifically @jelletreep), building on some older legacy code. 

> Is the use of these scripts just for the running of CI/CD? (i.e. don't play a role in building the package, or aren't called elsewhere in the codebase?)

Yes, I only think for CI/CD. The packaging is done at https://github.com/conda-forge/parcels-feedstock for anaconda, and at https://github.com/OceanParcels/parcels/blob/master/.github/workflows/pypi-release.yaml for PyPi

> I'm unfamiliar with some of the stuff happening in some of the scripts ([allowing for a plotting display??](https://github.com/VeckoTheGecko/parcels/blob/56e988ae0f627ec8994c656f5f4bab71154a2e47/.github/scripts/linux_integrationtest.sh#L5-L8), [custom variables being set](https://github.com/VeckoTheGecko/parcels/blob/56e988ae0f627ec8994c656f5f4bab71154a2e47/.github/scripts/osx_unittest.sh#L4-L5)). Could you shed light on these items?

I think these are remnants of when we also tested for plotting and visualisation. See https://github.com/OceanParcels/parcels/blob/master/tests/test_scripts.py. But I'm not entirely sure this actually works anymore?

**Reply by VeckoTheGecko** [reply]
Alrighty. I'll see if I can mock up what some updated workflows can look like, and we can discuss in a PR or something.

Even if the PR isn't accepted, this is good experience for me anyway.
